src/Common/Telemetry/Migration.php

```diff
@@ -0,0 +1,320 @@
+<?php
+/**
+ * Handles Telemetry migration from Freemius.
+ *
+ * @since   5.0.17
+ *
+ * @package TEC\Common\Telemetry
+ */
+namespace TEC\Common\Telemetry;
+
+use Automattic\WooCommerce\Utilities\ArrayUtil;
+use TEC\Common\StellarWP\Telemetry\Config;
+use TEC\Common\StellarWP\Telemetry\Opt_In\Opt_In_Subscriber;
+
+/**
+ * Class Migration
+ *
+ * @since   5.0.17
+
+ * @package TEC\Common\Telemetry
+ */
+final class Migration {
+	/**
+	 * The key we back up fs_accounts data to.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $fs_accounts_slug = 'tec_freemius_accounts_archive';
+
+	/**
+	 * The key we back up fs_active_plugins data to.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $fs_plugins_slug = 'tec_freemius_plugins_archive';
+
+	/**
+	 * Placeholder for fs_accounts data
+	 *
+	 * @since TBD
+	 *
+	 * @var [type]
+	 */
+	private $fs_accounts;
+
+	/**
+	 * List of our plugins to check for.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	public $our_plugins = [
+		'the-events-calendar',
+		'event-tickets'
+	];
+
+	/**
+	 * Determine if we are opted-in to Freemius
+	 *
+	 * @since 5.0.17
+	 *
+	 * @return boolean
+	 */
+	public function is_opted_in(): bool {
+		global $wpdb;
+		$fs_accounts = $wpdb->get_results( "SELECT `option_value` FROM $wpdb->options WHERE `option_name` = 'fs_accounts'", ARRAY_A );
+
+		if ( empty( $fs_accounts ) || $fs_accounts instanceof \WP_Error ) {
+			return false;
+		}
+
+		$fs_accounts = array_pop( $fs_accounts );
+
+		// Prevent issues with incomplete classes
+		$result = preg_replace(
+			'/O:(\d+):"(?:[^:]+)":/m',
+			'a:',
+			$fs_accounts['option_value']
+		);
```

Comment:
After this replacement, it's safe to unserialize, so you dont have to do string manipulation anymore.